core/src/banking_stage.rs

```diff
@@ -1315,9 +1295,9 @@ impl BankingStage {
                 "find_and_send_votes",
             );
             execute_and_commit_timings.find_and_send_votes_us = find_and_send_votes_time.as_us();
-            commit_time_us
+            (commit_time_us, commit_transaction_statuses)
         } else {
-            0
+            (0, vec![])
```

Comment:
Would returning an empty vector is a divergence from current behavior if
1. `executed_transactions_count` is `0`
2. `execution_results` is not empty right

It seems in this situation the current code we would return a `transactions_execute_and_record_status` constructed by

```
 let mut transactions_execute_and_record_status: Vec<bool> = execution_results
            .iter()
            .map(|execution_result| execution_result.was_executed())
            .collect();
```

which would be a vector of all false's with length `execution_results.len()`.

Reply:
Ah yeah, great catch. Fixed

Reply:
yeah this bit of code is so fragile, would be good to add some unit tests to cover the expected return values in some of these paths

Reply:
Cool, added a test for that path and the debug assert should cover the rest through the existing tests